lib/msal-node/src/network/HttpClient.ts

```diff
@@ -169,9 +171,10 @@ const networkRequestViaProxy = <T>(
                 });
                 const headers = Object.fromEntries(entries);
 
+                const parsedHeaders = headers as Record<string, string>;
                 const networkResponse: NetworkResponse<T> = {
-                    headers: headers as Record<string, string>,
-                    body: JSON.parse(body) as T,
+                    headers: parsedHeaders,
```

Comment:
nit: A util API to parse the response into the `networkResponse` format as this code is repeated?